enhancements/agent-installer/automated-workflow-for-agent-based-installer.md

```diff
@@ -0,0 +1,616 @@
+the network, and especially configuring hosts to boot them. This gives
+users the flexibility to perform those operations in any way they
+like, with whatever tool they prefer. A design for an all encompassing
+approach based on tools provided entirely by Red Hat is left for a
+later enhancement.
+
+The `agent create-image` command will need file-based inputs to
+describe the cluster and node-specific configuration. Using the
+existing file and custom resource types allows us to use the same
+version management techniques that we do for the APIs within a
+cluster.  We will use the `install-config.yaml` format for
+cluster-wide settings. We will use a new file, tentatively named
+`agent-config.yaml`, for settings that apply to only agent-based
+deployments.
+
+We will use manifests containing
+[NMStateConfig](https://github.com/openshift/assisted-service/blob/master/api/v1beta1/nmstate_config_types.go)
+resources to specify static network settings for the hosts. Although
+there are similar fields in `install-config.yaml`, they are only
+available when the platform setting is `baremetal`, and the
+agent-based installer must support other platforms.
+
+We anticipate using
+[Butane](https://github.com/coreos/butane/blob/main/docs/config-openshift-v4_10.md)
+file in the future for specifying storage settings, although that
+decision is subject to change in a future enhancement.
+
+The `agent create-image` command will produce a single image as output, to
+save storage space and reduce management complexity. The assisted
+installer service can already generate images with boot scripts that
+recognize which host is booting and apply host-specific network
+settings and roles (such as running different services) accordingly,
+and we will reuse that technique (and code).
+
+The image generated by the `agent create-image` command will be an RHCOS
+live ISO with embedded Ignition config containing systemd units to
+start the assisted service, agent, network, etc. running in containers
+pulled from the OpenShift release payload. Using RHCOS ensures that
+the same OS that will run OpenShift boots on the hosts. Using
+containers in the release payload means that when the payload is
+mirrored all of the tools needed to run the agent based deployment
+workflows will be present in the mirror.  More details about the
+various services follow below.
+
+The image generated by the `agent create-image` command will also
+contain the data in the input `install-config.yaml`,
+`agent-config.yaml`, and other files. That data may be converted to
+another format (we anticipate using the ZTP CRDs, since those map to
+the assisted-service API calls). The intermediate format will be
+considered another layer of control for the cluster creator, just as
+the manifests written by the `create manifests` sub-command are. The
+cluster creator will be able to generate those intermediate files with
+the sub-command `agent create-manifests`.
```

Comment:
I don't think we should block on this, but we will probably want to find another term since the manifests created by `agent create manifests` have a wholly different role to those created by the regular `create manifests`.

Reply:
Let's work that out with the installer team when we write that command and then come back and update this enhancement.